src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,76 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    private final MachineProducer truckProducer = new MachineProducer() {
```

Comment:
create implementation of producers like BulldozerProducer and implement method get()